### PR TITLE
fix: nit lwma_diff division check

### PR DESCRIPTION
**Pull request by jorgeantonio21**
Description
---
Add check for a possible edge case in the `lwma_diff` division check.

Motivation and Context
---
Auditing code.

How Has This Been Tested?
---

What process can a PR reviewer use to test or verify this change?
---

Breaking Changes
---

- [x] None
- [ ] Requires data directory on base node to be deleted
- [ ] Requires hard fork
- [ ] Other - Please specify

### CODE REVIEW

**Review by stringhandler — APPROVED** [review]
Not technically possible at the moment, but could be if the code was edited later

File: base_layer/core/src/proof_of_work/lwma_diff.rs [hunk 1]
```diff
@@ -87,7 +87,15 @@ impl LinearWeightedMovingAverage {
             .skip(1)
             .fold(0u128, |difficulty, (_, d)| difficulty + u128::from(d.as_u64()));
 
-        let ave_difficulty = difficulty / n;
+        let ave_difficulty = if let Some(ave) = difficulty.checked_div(n) {
```
Comment (sdbondi) [inline base_layer/core/src/proof_of_work/lwma_diff.rs:90]: `n` has already been checked above and can never be 0. `n = self.target_difficulties.len() - 1` where  `self.target_difficulties.len()` must be at least 2.

Comment (jorgeantonio21) [reply base_layer/core/src/proof_of_work/lwma_diff.rs:90]: Right, I didn't realize that. So these changes are not needed actually :)
